applicatie/logic/aggregeren.py

```diff
@@ -0,0 +1,231 @@
+                            if pval == '':
+                                pval = 'Totaal ' + key
+                            geteld_record.update({key: pval})
+                        elif key in vastedimensies:
+                            # voeg overige 'vaste' keys toe met de eigen code
+                            geteld_record.update({key: val})
+                        elif key == 'numgeteld':
+                            # houd bij hoevaak het record is geteld
+                            geteld_record.update({key: val+1})
+                    if 'numgeteld' not in geteld_record:
+                        geteld_record.update({'numgeteld': 1})
+            else:
+                # code komt niet voor in de codelijst
+                # N.B. dit zou eigenlijk niet mogen voorkomen
+                # omdat we eerder een codelijst check hebben uitgevoerd
+                if 'numgeteld' not in record:
+                    record_kort = dict()
+                    for k, v in record.items():
+                        if k == dimensie or k in vastedimensies:
+                            record_kort.update({k: v})
+                    telfouten.append("Code '{}' in record {} niet gevonden in codelijst.".format(val, str(record_kort)))
+            return geteld_record, telfouten
+
+        if record:
+            # voeg record toe aan de lijst
+            recs.append(record)
+            # tel record 1 niveau omlaag (= naar bovenliggend record) voor de gekozen dimensie
+            record, telfouten = tel_record(record, dimensie, vastedimensies, aggniveau, cds, niv, ptc)
+            # voeg eventuele fouten toe
+            errs.extend(telfouten)
+            # we gaan 1 niveau omlaag (= omhoog in de codelijst)
+            aggniveau = aggniveau - 1
+            # aggregeer het bovenliggende record (recursie)
+            aggregeren_record(recs, errs, record, dimensie, vastedimensies, aggniveau, cds, niv, ptc)
+
+        return recs, errs
+
+    # dit is de main loop waarin de door de data lopen
+    for index, record in enumerate(data):
+        # ieder record uit data aggregeren
+        aggrecords, fouten_agg = aggregeren_record([], [], record, dimensie, vastedimensies, aggniveau, cds, niv, ptc)
+        if aggrecords:
+            if len(aggrecords) == 1:
+                # indien 1 record terug, dan is deze 'niet geteld'
+                aggrecords[0].update({'geteld': False})
+            else:
+                # indien meer dan 1 record terug, dan zijn deze 'geteld'
+                # tenzij deze in een eerdere aggregatie 'niet geteld' zijn
+                for rec in aggrecords:
+                    if 'geteld' not in rec:
+                        rec.update({'geteld': True})
+                    else:
+                        geteld = rec.get('geteld')
+                        # record alleen als 'geteld' aanmerken
+                        # indien eerder ook geteld (beide True)
+                        geteld = geteld and True
```

Comment:
```x and True```

is logisch altijd gelijk aan x?

Reply:
geteld is True of False.
dus het resultaat is alleen True als geteld True is.